cmd/eudico-stats/stats.go

```diff
@@ -0,0 +1,229 @@
+//	ctx context.Context,
+//	node *SubnetDAGNode,
+//	subnetOutput *sca.SubnetOutput,
+//	relationShipChange *SubnetRelationshipChange,
+//	subnetNodeChange *NodeChange,
+//) (bool, *SubnetDAGNode, error) {
+//	log.Infow("processing subnet", "id", subnetOutput.Subnet.ID)
+//
+//	isUpdated := false
+//
+//	if node == nil {
+//		isUpdated = true
+//		node = &SubnetDAGNode{ID: subnetOutput.Subnet.ID, Children: make(map[address.SubnetID]*SubnetDAGNode)}
+//		log.Infow("is nil", "check", node.Children == nil, "id", node.ID)
+//	}
+//
+//	subnets, err := e.SubnetAPI.ListSubnets(ctx, subnetOutput.Subnet.ID)
+//	if err != nil {
+//		log.Errorw("subnets cannot be listed at height", "subnet", subnetOutput.Subnet.ID, "err", err)
+//		//return false, node, nil
+//		subnets = make([]sca.SubnetOutput, 0)
+//	}
+//
+//	log.Infow("listed subnets", "id", subnetOutput.Subnet.ID, "subnets", subnets, "nodeId", node.ID)
+//
+//	newChildrenMap := convertToMap(subnets)
+//
+//	// TODO: recheck node remove logic
+//	for child, _ := range node.Children {
+//		if _, ok := newChildrenMap[child]; !ok {
+//			relationShipChange.Remove(node.ID, child)
+//			delete(node.Children, child)
+//			subnetNodeChange.Remove(child)
+//		}
+//	}
+//
+//	for child, output := range newChildrenMap {
+//		if _, ok := node.Children[child]; !ok {
+//			relationShipChange.Add(node.ID, child)
+//		}
+//		var childNode *SubnetDAGNode
+//		log.Infow("before is nil", "check", node.Children == nil, "id", node.ID, "child", child, "subnetOutput", subnetOutput.Subnet.ID)
+//		isUpdated, childNode, err = e.traverseSubnet(ctx, node.Children[child], &output, relationShipChange, subnetNodeChange)
+//		if err != nil {
+//			continue
+//		}
+//		log.Infow("is nil", "check", node.Children == nil, "id", node.ID, "child", child, "subnetOutput", subnetOutput.Subnet.ID)
+//		node.Children[child] = childNode
+//	}
+//
+//	// first check if the node itself has changed
+//	e.updateSubnetStats(*subnetOutput, len(newChildrenMap), subnetNodeChange)
+//
+//	return isUpdated, node, nil
+//}
+//
+//func (e *EudicoStats) updateSubnetStats(subnetOutput sca.SubnetOutput, subnetsCount int, subnetNodeChange *NodeChange) {
+//	newNode := newSubnetNode(subnetOutput, subnetsCount)
+//	old, ok := e.subnetStats[subnetOutput.Subnet.ID]
+//	if !ok {
+//		log.Infow("new subnet created", "id", subnetOutput.Subnet.ID)
+//		e.subnetStats[subnetOutput.Subnet.ID] = newNode
+//		subnetNodeChange.Add(subnetOutput.Subnet.ID)
+//		return
+//	}
+//
+//	if !old.Equals(&newNode) {
+//		log.Infow("new subnet updated", "id", subnetOutput.Subnet.ID)
+//		e.subnetStats[subnetOutput.Subnet.ID] = newNode
+//		subnetNodeChange.Add(subnetOutput.Subnet.ID)
+//		return
+//	}
+//}
+//
+//func newSubnetNode(output sca.SubnetOutput, count int) SubnetNode {
+//	return SubnetNode{
+//		SubnetID:    output.Subnet.ID,
+//		SubnetCount: count,
+//		Consensus:   output.Consensus,
+//		Subnet:      output.Subnet,
+//		Version:     0,
+//	}
+//}
```

Comment:
clean?

Reply:
oh yeah, this is the old method I used for reference. I guess it's not much of a use now.